server/s3_bucket_metadata.cc

```diff
@@ -220,9 +223,132 @@ int S3BucketMetadata::from_json(std::string content) {
+    }
+
+    if (!rule_object["Priority"].isNull()) {
+      rule_child_str = rule_object["Priority"].asString();
+      xml_str += "<Priority>" + rule_child_str + "</Priority>";
+    }
+
+    if (!rule_object["ID"].isNull()) {
+      rule_child_str = rule_object["ID"].asString();
+      xml_str += "<ID>" + rule_child_str + "</ID>";
+    }
+
+    if (!rule_object["DeleteMarkerReplication"].isNull()) {
+
+      rule_child_str =
+          rule_object["DeleteMarkerReplication"]["Status"].asString();
+      xml_str += "<DeleteMarkerReplication><Status>" + rule_child_str +
+                 "</Status></DeleteMarkerReplication>";
+    }
+
+    if (!rule_object["Destination"].isNull()) {
+
+      rule_child_str = rule_object["Destination"]["Bucket"].asString();
+      xml_str +=
+          "<Destination><Bucket>" + rule_child_str + "</Bucket></Destination>";
+    }
+
+    std::string key_str, val_str, pre_str;
+
+    if (!rule_object["Filter"].isNull()) {
+      if (!rule_object["Filter"]["And"]["Tag"].isNull() &&
+          rule_object["Filter"]["And"]["Prefix"].isNull()) {
+        // If tag,and nodes are present in filter and Prefix is not present.
+
+        xml_str += "<Filter><And>";
+        Json::Value tag_array = rule_object["Filter"]["And"]["Tag"];
+        for (unsigned int index = 0; index < tag_array.size(); ++index) {
+
+          Json::Value tag_object = tag_array[index];
+          key_str = tag_object["Key"].asString();
+          val_str = tag_object["Value"].asString();
+
+          xml_str += "<Tag><Key>" + key_str + "</Key><Value>" + val_str +
+                     "</Value></Tag>";
+        }
+        xml_str += "</And></Filter>";
+      } else if (!rule_object["Filter"]["And"]["Tag"].isNull() &&
+                 !rule_object["Filter"]["And"]["Prefix"].isNull()) {
+        // If tag,and,Prefix nodes are present  in filter
+
+        xml_str += "<Filter><And><Prefix>";
+        pre_str = rule_object["Filter"]["And"]["Prefix"].asString();
+
+        xml_str += pre_str + "</Prefix>";
+        Json::Value tag_array = rule_object["Filter"]["And"]["Tag"];
+        for (unsigned int index = 0; index < tag_array.size(); ++index) {
+          Json::Value tag_object = tag_array[index];
+          key_str = tag_object["Key"].asString();
+          val_str = tag_object["Value"].asString();
+
+          xml_str += "<Tag><Key>" + key_str + "</Key><Value>" + val_str +
+                     "</Value></Tag>";
+        }
+        xml_str += "</And></Filter>";
+      } else if (!rule_object["Filter"]["Tag"].isNull()) {
+        // If only tag node is present in filter
+
+        key_str = rule_object["Filter"]["Tag"]["Key"].asString();
+        val_str = rule_object["Filter"]["Tag"]["Value"].asString();
+        xml_str += "<Filter><Tag><Key>" + key_str + "</Key><Value>" + val_str +
+                   "</Value></Tag></Filter>";
+      } else if (!rule_object["Filter"]["Prefix"].isNull()) {
+        // If only Prefix node is present in filter
```

Comment:
Can we add a comment somewhere here to say we only support Filter/Prefix, not Prefix (deprecated)